src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.NamedRequest.cs

```diff
@@ -0,0 +1,1070 @@
+                firstResult.Events.Should().NotContainErrors();
+
+
+                var secondCode = $$$"""
+            GET https://httpbin.org/headers
+            Detectedserver: {{binHeader.response.headers.Accept}}
+            ###
+            """;
+
+                var secondResult = await kernel.SendAsync(new SubmitCode(secondCode));
+                var diagnostics = secondResult.Events.Should().ContainSingle<DiagnosticsProduced>().Which;
+                diagnostics.Diagnostics.First().Message.Should().Be($$$"""The supplied header name 'Accept' does not exist in the named request.""");
+            }
+
+            [Theory]
+            [InlineData("json.response.body.$.slideshow.slides.title", "Wake up to WonderWidgets!")]
+            [InlineData("json.response.body.$.slideshow.slides.type", "all")]
+            public async Task json_with_additional_syntax_depths_can_be_accessed_correctly(string path, string end)
+            {
+                // Request Variables
+                // Request variables are similar to file variables in some aspects like scope and definition location.However, they have some obvious differences.The definition syntax of request variables is just like a single-line comment, and follows // @name requestName or # @name requestName just before the desired request url. 
+
+                var responseHandler = new InterceptingHttpMessageHandler((message, _) =>
+                {
+                    var response = new HttpResponseMessage(HttpStatusCode.OK);
+                    response.RequestMessage = message;
+                    var contentString = @"
+                        {
+              ""slideshow"": {
+                ""author"": ""Yours Truly"",
+                ""date"": ""date of publication"",
+                ""slides"": [
+                  {
+                    ""title"": ""Wake up to WonderWidgets!"",
+                    ""type"": ""all""
+                  },
+                  {
+                    ""items"": [
+                      ""Why <em>WonderWidgets</em> are great"",
+                      ""Who <em>buys</em> WonderWidgets""
+                    ],
+                    ""title"": ""Overview"",
+                    ""type"": ""all""
+                  }
+                ],
+                ""title"": ""Sample Slide Show""
+              }
+            }
+            ";
+                    response.Content = new StringContent(contentString, Encoding.UTF8, "application/json");
+                    return Task.FromResult(response);
+                });
+                var client = new HttpClient(responseHandler);
+                using var kernel = new HttpKernel("http", client);
+
+                var firstCode = """
+            @baseUrl = https://httpbin.org/json
+
+            # @name json
+            GET {{baseUrl}}
+            Content-Type: application/json
+
+            ###
+            """;
```

Comment:
Indenting raw strings so their left margin is indented further than the main code will improve readability, e.g.: 

```csharp
                var firstCode = """
                    @baseUrl = https://httpbin.org/json

                    # @name json
                    GET {{baseUrl}}
                    Content-Type: application/json

                    ###
                    """;

```